scripts/cluster-install/install.py

```diff
@@ -0,0 +1,392 @@
+
+
+def _prepare_postgresql_config_files(instances_dict):
+    cluster_nodes = [instances_dict['postgresql_nodes'][j].private_ip for
+                     j in range(3)]
+    for node in instances_dict['postgresql_nodes']:
+        with open('postgresql_config.yaml') as f:
+            config_file = yaml.load(f, yaml.Loader)
+        conf_file_name = _write_crt_to_config(config_file, node.name,
+                                              'postgresql_server')
+        config_file['postgresql_server']['cluster']['nodes'] = cluster_nodes
+        with open(conf_file_name, 'w') as f:
+            yaml.dump(config_file, f)
+
+
+def _rabbitmq_credential_generator():
+    return ''.join(random.choice(string.ascii_lowercase + string.digits) for
+                   _ in range(6))
+
+
+def _prepare_rabbitmq_config_files(instances_dict):
+    cluster_members = {instances_dict['rabbitmq_nodes'][j].name: {
+        'default': instances_dict['rabbitmq_nodes'][j].private_ip}
+        for j in range(3)}
+    first_rabbitmq = instances_dict['rabbitmq_nodes'][0]
+    rabbitmq_username = _rabbitmq_credential_generator()
+    rabbitmq_password = _rabbitmq_credential_generator()
+    for i, node in enumerate(instances_dict['rabbitmq_nodes']):
+        with open('rabbitmq_config.yaml') as f:
+            config_file = yaml.load(f, yaml.Loader)
+        config_file['rabbitmq']['username'] = rabbitmq_username
+        config_file['rabbitmq']['password'] = rabbitmq_password
+        config_file['rabbitmq']['cluster_members'] = cluster_members
+        conf_file_name = _write_crt_to_config(config_file, node.name,
+                                              'rabbitmq')
+        config_file['rabbitmq']['nodename'] = node.name
+        if i != 0:
+            config_file['rabbitmq']['join_cluster'] = first_rabbitmq.name
+        with open(conf_file_name, 'w') as f:
+            yaml.dump(config_file, f, default_flow_style=False)
+    return rabbitmq_username, rabbitmq_password
+
+
+def _create_ssl_inputs(node_name):
+    ssl_inputs = {}
+    cert = REMOTE_INSTALL_CLUSTER + '/certs/{}_cert.pem'.format(node_name)
+    key = REMOTE_INSTALL_CLUSTER + '/certs/{}_key.pem'.format(node_name)
+    ca = REMOTE_INSTALL_CLUSTER + '/certs/ca.pem'
+    cert_path_list = ['internal_cert_path', 'external_cert_path',
+                      'postgresql_client_cert_path']
+    key_path_list = ['internal_key_path', 'external_key_path',
+                     'postgresql_client_key_path']
+    ca_path_list = ['ca_cert_path', 'external_ca_cert_path']
+    for cert_path in cert_path_list:
+        ssl_inputs[cert_path] = cert
+    for key_path in key_path_list:
+        ssl_inputs[key_path] = key
+    for ca_path in ca_path_list:
+        ssl_inputs[ca_path] = ca
+    return ssl_inputs
+
+
+def _prepare_manager_config_files(instances_dict, rabbitmq_credentials):
+    postgresql_nodes = [instances_dict['postgresql_nodes'][j].private_ip for
+                        j in range(3)]
```

Comment:
i'm not sure why this function uses ranges+indexes (two places in this func, several more in the pr) instead of just iterating over `instances_dict['postgresql_nodes']`?

Reply:
Great question, the reason for it is that `instances_dict['postgresql_nodes'] `is a list of VMs objects, whereas we are looking for their IPs.